src/Features/Core/Portable/CodeFixes/CodeFixService.cs

```diff
@@ -183,51 +183,33 @@ internal partial class CodeFixService : ICodeFixService
                 if (diagnostic.IsSuppressed)
                     continue;
 
-                cancellationToken.ThrowIfCancellationRequested();
-
-                aggregatedDiagnostics.GetOrAdd(diagnostic.GetTextSpan(), _ => new List<DiagnosticData>()).Add(diagnostic);
+                var list = aggregatedDiagnostics.GetOrAdd(diagnostic.GetTextSpan(), static _ => new List<DiagnosticData>());
+                list.Add(diagnostic);
             }
 
             if (aggregatedDiagnostics.Count == 0)
                 return ImmutableArray<CodeFixCollection>.Empty;
 
             // Order diagnostics by DiagnosticId so the fixes are in a deterministic order.
-            foreach (var diagnosticsWithSpan in aggregatedDiagnostics.Values)
-            {
-                diagnosticsWithSpan.Sort(s_diagnosticDataComparisonById);
-            }
+            foreach (var (_, diagnosticList) in aggregatedDiagnostics)
+                diagnosticList.Sort(s_diagnosticDataComparisonById);
 
             // append fixes for all diagnostics with the same diagnostics span
-            using var resultDisposer = ArrayBuilder<CodeFixCollection>.GetInstance(out var result);
+            using var _1 = ArrayBuilder<CodeFixCollection>.GetInstance(out var result);
 
             // 'CodeActionRequestPriority.Lowest' is used when the client only wants suppression/configuration fixes.
             if (priority != CodeActionRequestPriority.Lowest)
             {
-                foreach (var spanAndDiagnostic in aggregatedDiagnostics)
-                {
-                    await AppendFixesAsync(
-                        document, spanAndDiagnostic.Key, spanAndDiagnostic.Value, fixAllForInSpan: false,
-                        priority, options, result, addOperationScope, cancellationToken).ConfigureAwait(false);
-                }
-            }
-
-            if (result.Count > 0 && TryGetWorkspaceFixersPriorityMap(document, out var fixersForLanguage))
-            {
-                // sort the result to the order defined by the fixers
-#pragma warning disable IDE0007 // Use implicit type - Explicit type is need to suppress an incorrect nullable warning on dereferencing the map.
-                ImmutableDictionary<CodeFixProvider, int> priorityMap = fixersForLanguage.Value;
-#pragma warning restore IDE0007 // Use implicit type
-                result.Sort((d1, d2) => GetValue(d1).CompareTo(GetValue(d2)));
-
-                int GetValue(CodeFixCollection c)
-                    => priorityMap.TryGetValue((CodeFixProvider)c.Provider, out var value) ? value : int.MaxValue;
+                await AppendFixesAsync(
+                    document, aggregatedDiagnostics, fixAllForInSpan: false,
+                    priority, options, result, addOperationScope, cancellationToken).ConfigureAwait(false);
```

Comment:
this is the meat of hte change.  previously we would go through each span/diagnostic asked, adn then (internally) go through all fixers to get the fixes for them.  Then, after that we'd come and sort the results by codefixprovider to get the final result.  This meant we couldn't return anything until they had all computed and we then ordered by fixer.

Now, we just order by fixers first, and run each fixer against the span/diags they can fix.